### PR TITLE
feat: checksum downloaded files

### PR DESCRIPTION
**Pull request by smira**
For each downloaded, verify SHA256 and SHA512 checkums of the files.
This ensures that source files haven't been tampered with.

Two checksums to make sure that even if one of the hashing algorithms is
compromised, another one guarantees authenticity.

Script `generate-checksums.sh` to generate checksum files.

Some GitHub download URLs were rewritten to equivalent URLs which
provide nice filename for `wget`.

Linux kernel step switched to use common `download` function.

Curl CA certificates bundle now additionally verifies sha256 checksum
downloaded from the site. CA certificate bundle is not verified via
committed checksum file as it might change upstream on its own schedule
we can't control.

### CODE REVIEW

File: SHA512SUM [hunk 1]
```diff
@@ -0,0 +1,2 @@
+0fca326feb1d5f5fe505a827b20237fe3ec9c13eaf7ec7e35847fd71184f605ba1cefe1314b1b8f8a29c0aa9d88162849ee1c1a3e70c2f7407d88339b17edb30  binutils-2.31.1.tar.xz
```
Comment (smira) [inline SHA512SUM]: I can also split checksums by folder where it is being used...

**Comment by andrewrynhard** [comment]
@smira Awesome stuff. I'm just curious, what's the reasoning for sha256 and sha512 checksums. Why both instead of just one. Not opposed to both, just curious.

**Comment by smira** [comment]
@andrewrynhard that comes from my experience with say Debian package repositories - they use multiple checksums, so that even if one of them gets compromised, another one would guarantee that check was valid. Probably better to use something from different hash families, but I'm not sure what's the another one which is easily available for us to use?

**Comment by smira** [comment]
so in other words it's much harder to compromise two different hash functions at the same time for the same file, even if we assume it's doable for one hash function (like it happened with md5)

**Comment by andrewrynhard** [comment]
@smira Excellent. Sounds good to me.

**Comment by smira** [comment]
PR updated, now it builds correctly, I'm testing with `talos` now...

**Comment by smira** [comment]
Talos build is okay, ready for the review

**Pull request merged**